packages/kit/src/runtime/client/utils.js

```diff
@@ -23,72 +24,130 @@ export function scroll_state() {
 	};
 }
 
-/** @param {Event} event */
-export function find_anchor(event) {
+const warned = new WeakSet();
+
+/**
+ * @param {Element} element
+ * @param {string} name
+ * @param {string | null} value
+ * @param {string[]} options
+ */
+function validate(element, name, value, options) {
+	if (warned.has(element)) return;
+	if (value === null) return;
+	if (!options.includes(value)) {
+		warned.add(element);
+		console.error(
+			`Unexpected value for ${name} — should be one of ${options
+				.map((option) => JSON.stringify(option))
+				.join(', ')}`,
+			element
+		);
+	}
+}
+
+/** @type {Record<string, number>} */
+const levels = {
+	tap: PRIORITY_TAP,
+	hover: PRIORITY_HOVER,
+	viewport: PRIORITY_VIEWPORT,
+	page: PRIORITY_PAGE,
+	'': PRIORITY_HOVER,
+	off: -1
+};
+
+/**
+ * @param {Element} element
+ * @param {string} base
+ */
+export function find_anchor(element, base) {
```

Comment:
We should really document the return type of this and tag it with `@returns` -- it's got a gnarly return object that could change subtly in the future and bite us.

Reply:
gonna hold off on that because all this stuff will be refactored as part of #7828